app/Controllers/Donation/NewDonationController.php

```diff
@@ -19,12 +19,9 @@ public function index( FunFunFactory $ffFactory, Request $request ): Response {
 
 		$amount = intval( $request->get( 'amount', 0 ) );
 		$paymentType = (string)$request->get( 'paymentType', '' );
-		$interval = $request->get( 'interval', 0 );
-		if ( $interval !== null ) {
-			$interval = intval( $interval );
-		}
+		$interval = $this->getIntervalFromRequest( $request );
 
-		$validationResult = $ffFactory->newPaymentValidator()->validatePaymentData( $amount, $interval, $paymentType, $ffFactory->newDonationPaymentValidator() );
+		$validationResult = $ffFactory->newPaymentValidator()->validatePaymentData( $amount, $interval ?? 0, $paymentType, $ffFactory->newDonationPaymentValidator() );
```

Comment:
while reviewing I was wondering what that 0 here stands for as a fallback value.
Do we express with this that we want "one time payment" as a fallback user selection?

If that's correct, maybe we could give that 0 a name to not have it in there as a magic number.